actix-tracing/src/lib.rs

```diff
@@ -0,0 +1,302 @@
+//! Actix tracing - support for tokio tracing with Actix services.
+#![deny(rust_2018_idioms, warnings)]
+
+use std::marker::PhantomData;
+use std::task::{Context, Poll};
+
+use actix_service::{apply, IntoServiceFactory, Service, ServiceFactory, Transform};
+use futures_util::future::{ok, Either, Ready};
+use tracing_futures::{Instrument, Instrumented};
+
+/// Trait representing a type that is able to produce spans for a given value.
+///
+/// The generic argument `T` is typically the request type of a `Service`
+/// implementation.
+pub trait MakeSpan<T> {
+    /// Creates a new span for the given request value.
+    ///
+    /// By default, this function returns `None`.
+    fn span_for(&self, _req: &T) -> Option<tracing::Span> {
```

Comment:
should be required method, no reason for default impl

Reply:
Agree.